app/src/org/commcare/activities/HomeScreenBaseActivity.java

```diff
@@ -374,16 +374,23 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
                         finish();
                     } else if (resultCode == DeveloperPreferences.RESULT_SYNC_CUSTOM) {
                         try {
-                            Uri uri = intent.getData();
-                            String filePath = UriToFilePath.getPathFromUri(CommCareApplication.instance(), uri);
-                            if (filePath != null) {
-                                File f = new File(filePath);
-                                if (f != null && f.exists()) {
-                                    formAndDataSyncer.performCustomRestoreFromFile(this, f);
+                            String filePath = DeveloperPreferences.getCustomRestoreDocLocation();
```

Comment:
Would be nice to pull this chunk of code out into its own method, since there's now a lot going on here